library/data/registry_dicts/src/main/java/org/quiltmc/qsl/registry/dict/api/WrongValueClassException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.dict.api;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+/**
+ * Thrown by {@link RegistryDict#get(Registry, Identifier, Class)}, should a dictionary with a matching ID be found
+ * but have a different value class than expected.
+ */
```

Comment:
This class seems unnecessary. In the situations where it would be raised, I'd just throw a runtime error and crash. I can't think of another good, legitimate response to this kind of error.